dist/tools/programmer/programmer.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2021 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+import time
+import shlex
+import subprocess
+import argparse
+
+from contextlib import contextmanager
+
+
+SUCCESS = "\033[32;1m✓\033[0m"
+FAILED = "\033[31;1m×\033[0m"
+SPIN = ["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]
+
+
+class Programmer:
+
+    @contextmanager
+    def spawn_process(self):
+        """Yield a subprocess running in background."""
+        kwargs = {} if self.verbose else {
+            "stdout": subprocess.PIPE,
+            "stderr": subprocess.PIPE
+        }
+        yield subprocess.Popen(shlex.split(self.cmd), **kwargs)
+
+    def spin(self, process):
+        """Print a spinning icon while programmer process is running."""
+        while process.poll() is None:
+            for index in range(len(SPIN)):
+                sys.stdout.write(
+                    "\r \033[36;1m{}\033[0m {} in progress "
+                    "(programmer: '{}')"
+                    .format(SPIN[index], self.action, self.programmer)
+                )
+                sys.stdout.flush()
+                time.sleep(0.1)
+
+    def print_status(self, process, elapsed):
+        """Print status of background programmer process."""
+        print(
+            "\r \u001b[2K{} {} {} (programmer: '{}' - duration: {:0.2f}s)"
```

Comment:
Maybe add a `(for full output set PROGRAMMER_QUIET=0)` here

Reply:
Good idea, I added something